todo_app/app.py

```diff
@@ -104,6 +141,98 @@ def move_to_todo_item():            # Called to move a 'card' BACK to 'todo' (wa
       print(doc)
     return redirect("/")
 
+
+@app.route('/login/callback', methods = ["GET","POST"])
+def login():
+    #This is where github has it's call back sending to .. (set in github)
+
+
+    #NOTES FROM INSTRUCTORS , I JUST WANT TO KEEP HERE RIGHT NOW ..
+    
+    # we want to call the login user function .. import from flask login
+    # To read this stage we need to parse the info github has given up
+   # understand what has github given us .. ask github who is the user .. 
+   # github will give token .. we're gonna use that token to ask github
+   # who is this
+   # login that user    --- flask login library
+
+
+
+    # Get the access_token
+
+    url, headers, body = Clientsecurity.prepare_token_request(
+        "https://github.com/login/oauth/access_token",
+        authorization_response=request.url
+    )
+    print("REACHED HERE - CLIENT SECURITY INFO PRINTED BELOW:")
+    print(Clientsecurity.prepare_token_request(
+        "https://github.com/login/oauth/access_token",
+        authorization_response=request.url
+    ))
+    
+    token_response = requests.post(
+        url,
+        headers=headers,
+        data=body,
+        auth=(client_id, client_secret)
+        )
+
+    # Now to get the users data
+
+    Clientsecurity.parse_request_body_response(token_response.text)
+    url, headers, body = Clientsecurity.add_token("https://api.github.com/user")
+    
+    user_response = requests.get(url, headers=headers, data=body)
+    # Has to be read as a JSON
+    full_user_response = json.loads(user_response.text)
+    # Now has to be split into individual values
+    values_view = full_user_response.values()
+    # Now has to be put into an interative list
+    full_user_view = iter(values_view)
+    # Now we need the first thing off that list, which is the user name
+    the_user_name = next(full_user_view)
+
+    # Unbelievable .. login_user won't actually accept a string it
+    # will only take a user object as the first parameter, not a string
+    # representing the user
+
+    finalusername = User(the_user_name)   
```

Comment:
```suggestion
    the_user_name = user_response.json()['login']
```
The user_response looks like this:
```
{'login': 'hugh-emerson', 'id': 10462681, etc...}
```
After using .json() or json.loads, you have a dictionary. You have used dictionaries a lot  - you access values by `your_dictionary['key']`
You are currently relying on the order of the values in the object rather than just picking one by name, which would be far clearer and more reliable.